src/app/leaves/components/LeavesRequestList.tsx

```diff
@@ -0,0 +1,188 @@
+} from "@/components/ui/table";
+import useGroupedData from "@/hooks/useGroupedData";
+import { LeaveRequestsTypes } from "@/types";
+import { StandardPage } from "@/common/StandardPage/StandardPage";
+import { Plus as PlusIcon } from "lucide-react";
+import { LeavesStatusBadge } from "@/common/StatusBadge/LeavesStatusBadge";
+import LeaveTableActionRender from "./LeaveActionRender";
+
+const leaveRequests = [
+  {
+    id: 1,
+    leaveType: "Sick Leave",
+    startDate: "2024-11-01",
+    endDate: "2024-11-03",
+    status: "accept",
+    reasonForLeave: "Flu",
+    requestedBy: "John Doe",
+  },
+  {
+    id: 2,
+    leaveType: "Vacation",
+    startDate: "2024-12-10",
+    endDate: "2024-12-20",
+    status: "pending",
+    reasonForLeave: "Family vacation",
+    requestedBy: "Jane Smith",
+  },
+  {
+    id: 3,
+    leaveType: "Personal Leave",
+    startDate: "2024-11-15",
+    endDate: "2024-11-16",
+    status: "accept",
+    reasonForLeave: "Personal matter",
+    requestedBy: "Emily Johnson",
+  },
+  {
+    id: 4,
+    leaveType: "Maternity Leave",
+    startDate: "2024-12-01",
+    endDate: "2025-05-31",
+    status: "pending",
+    reasonForLeave: "Childbirth",
+    requestedBy: "Sarah Brown",
+  },
+  {
+    id: 5,
+    leaveType: "Unpaid Leave",
+    startDate: "2024-11-20",
+    endDate: "2024-11-25",
+    status: "reject",
+    reasonForLeave: "Travel",
+    requestedBy: "Michael Lee",
+  },
+];
+
+export const LeavesRequestList = () => {
+
+  const columns: ColumnDef<LeaveRequestsTypes>[] = useMemo(
+    () => [
+      {
+        accessorKey: "requestedBy",
+        header: "Requested By",
+        cell: ({ row }) => (
+          <div className="capitalize">{row.getValue("requestedBy")}</div>
+        ),
+      },
+      {
+        accessorKey: "leaveType",
+        header: "Leave Type",
+        cell: ({ row }) => (
+          <div className="capitalize">{row.getValue("leaveType")}</div>
+        ),
+      },
+      {
+        accessorKey: "startDate",
+        header: "Start Date",
+        cell: ({ row }) => <div>{row.getValue("startDate")}</div>,
+      },
+      {
+        accessorKey: "endDate",
+        header: "End Date",
+        cell: ({ row }) => <div>{row.getValue("endDate")}</div>,
+      },
+      {
+        accessorKey: "status",
+        header: "Status",
+        cell: ({ row }) => (
+          <LeavesStatusBadge status={row.getValue("status")} />
+        ),
+      },
+      {
+        id: "actions",
+        enableHiding: false,
+        header: () => {
+          return <span>Action</span>;
+        },
+        cell: ({ row }) => {
+          return (
+            <LeaveTableActionRender leaveRequestStatus={row.original.status} />
+          );
+        },
+      },
+    ],
+    []
+  );
+
+  const table = useReactTable({
+    data: leaveRequests,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  const groupedData = useGroupedData(leaveRequests, "shift");
+  const toNavigate = () => {
```

Comment:
This doens't look like a correct name. Function name should use verb. How about navigateToDetails or wherever this is going.